superset/viz.py

```diff
@@ -2062,6 +2062,40 @@ def get_data(self, df):
         }
 
 
+class DeckPolygon(BaseDeckGLViz):
+
+    """deck.gl's Polygon Layer"""
+
+    viz_type = 'deck_polygon'
+    verbose_name = _('Deck.gl - Polygon')
+    deser_map = {
+        'json': json.loads,
+        'polyline': polyline.decode,
+    }
+
+    def get_position(self, d):
+        return d[self.form_data.get('line_column')]
+
+    def query_obj(self):
+        d = super(DeckPolygon, self).query_obj()
+        line_col = self.form_data.get('line_column')
+        if d['metrics']:
+            d['groupby'].append(line_col)
+        else:
+            d['columns'].append(line_col)
+        return d
+
+    def get_properties(self, d):
+        fd = self.form_data
+        deser = self.deser_map[fd.get('line_type')]
```

Comment:
We could save one hash lookup per row (`get_properties` gets called for every row in the df) by setting `self.deser = self.deser_map[fd.get('line_type')]` in `__init__`. (I missed out on this simple optimization in DeckPath...)